c2rust-transpile/src/c_ast/conversion.rs

```diff
@@ -687,6 +719,11 @@ impl ConversionContext {
                         CTypeKind::Function(ret, arguments, is_variadic, is_noreturn, has_proto);
                     self.add_type(new_id, not_located(function_ty));
                     self.processed_nodes.insert(new_id, FUNC_TYPE);
+
+                    let pointer_ty = CTypeKind::Pointer(CQualTypeId::new(CTypeId(new_id)));
+                    let new_id = self.id_mapper.fresh_id();
+                    self.add_type(new_id, not_located(pointer_ty));
+                    self.processed_nodes.insert(new_id, OTHER_TYPE);
```

Comment:
What's this doing exactly?  Adding a fn ptr type?  And how come a fresh `new_id` is created here shadowing the original one?

Reply:
Yes. I'll add a comment explaining what/why. The second `new_id` is because we're also adding a second type; each type needs its own ID.